File: 4378-nearest-available-drone/nearest-available-drone.py
class Solution:
    def nearestDrone(self, drones: list[list[int]], target: list[int]) -> int:
        
        idx = -1
        mini = float('inf')

        a, b = target

        for i in range(len(drones)):
            x, y, z = drones[i]

            m_dist = abs(x - a) + abs(y - b)

            if m_dist <= z and m_dist < mini:
                mini = m_dist
                idx = i

        return idx

File: 4378-nearest-available-drone/nearest-available-drone.cpp
class Solution {
public:
    int nearestDrone(vector<vector<int>>& drones, vector<int>& target) {
        
    int idx = -1;
    int mini = INT_MAX;

    int a = target[0];
    int b = target[1];
    for(int i  = 0;i<drones.size();i++){

        int x = drones[i][0];
        int y = drones[i][1];
        int z = drones[i][2];

        int min_d = abs(x-a) + abs(y- b);

        if (min_d <= z and min_d < mini){

            mini = min_d;
            idx = i;
        }

    }   
    return idx;

    }
};